File: uikit/src/main/java/com/cometchat/pro/uikit/Reaction/model/Reaction.java
package com.cometchat.pro.uikit.Reaction.model;

import android.os.Parcel;
import android.os.Parcelable;

public class Reaction implements Parcelable {
    String name;
    String code;

    public Reaction(String name, String code) {
        this.name = name;
        this.code = code;
    }

    protected Reaction(Parcel in) {
        name = in.readString();
        code = in.readString();
    }

    public static final Creator<Reaction> CREATOR = new Creator<Reaction>() {
        @Override
        public Reaction createFromParcel(Parcel in) {
            return new Reaction(in);
        }

        @Override
        public Reaction[] newArray(int size) {
            return new Reaction[size];
        }
    };

    public String getName() {
        return name;
    }

    public String getCode() {
        return code;
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel parcel, int i) {
        parcel.writeString(name);
        parcel.writeString(code);
    }
}

File: uikit-kotlin/src/main/java/com/cometchat/pro/uikit/reaction/model/Reaction.kt
package com.cometchat.pro.uikit.reaction.model

import android.os.Parcel
import android.os.Parcelable

class Reaction() : Parcelable{
    lateinit var name: String
    var code = 0

    constructor(name: String, code: Int) : this(){
        this.name = name
        this.code = code
    }

    constructor(parcel: Parcel) : this() {
        name = parcel.readString()!!
        code = parcel.readInt()
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeString(name)
        parcel.writeInt(code)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object CREATOR : Parcelable.Creator<Reaction> {
        override fun createFromParcel(parcel: Parcel): Reaction {
            return Reaction(parcel)
        }

        override fun newArray(size: Int): Array<Reaction?> {
            return arrayOfNulls(size)
        }
    }

}